test/AccessTokenVerifier/AccessTokenVerifier.behaviour.ts

```diff
@@ -21,6 +21,40 @@ const shouldBehaveLikeAccessTokenVerifier = function () {
       expect(await this.auth.callStatic.verify(this.token, signature.v, signature.r, signature.s)).to.be.true;
     });
 
+    it("should return signer address", async function () {
+      const signature = splitSignature(await signAccessToken(this.signers.admin, this.domain, this.token));
+
+      await expect(
+        this.auth.verifySignerOf(this.token, signature.v, signature.r, signature.s, this.signers.admin.address),
+      ).to.not.be.reverted;
+      expect(
+        await this.auth.callStatic.verifySignerOf(
+          this.token,
+          signature.v,
+          signature.r,
+          signature.s,
+          this.signers.admin.address,
+        ),
+      ).to.equal(this.signers.admin.address);
+    });
+
+    it("should return 0 address", async function () {
+      const signature = splitSignature(await signAccessToken(this.signers.admin, this.domain, this.token));
+
+      await expect(
+        this.auth.verifySignerOf(this.token, signature.v, signature.r, signature.s, this.signers.user0.address),
+      ).to.not.be.reverted;
+      expect(
+        await this.auth.callStatic.verifySignerOf(
+          this.token,
+          signature.v,
+          signature.r,
+          signature.s,
+          this.signers.user0.address,
+        ),
+      ).to.equal(ethers.constants.Zero);
+    });
+
```

Comment:
Can possibly also add a test that returns an unexpected address when you pass in a wrong signature